# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1711563990 -3600
# Parent  4658805eb0d7b08ba8cdf1a5e2651767e8d7582a
Feature: Open Activity Manager in tab.

The Activity Manager is useful, but it's stand-alone window gets in the way.
Opening it in a tab by default avoids having to use the userChromeJS add-on.
* * *
Feature: Open Activity Manager in tab, introduce pref mail.activity.managerInTab.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1376,8 +1376,9 @@ pref("pdfjs.firstRun", true);
 pref("pdfjs.previousHandler.preferredAction", 0);
 pref("pdfjs.previousHandler.alwaysAskBeforeHandling", false);
 
 pref("mail.activity.loglevel", "Warn");
+pref("mail.activity.managerInTab", true);
 
 // The number of public recipients before we offer BCC addressing.
 pref("mail.compose.warn_public_recipients.threshold", 15);
 
diff --git a/mail/base/content/mailCore.js b/mail/base/content/mailCore.js
--- a/mail/base/content/mailCore.js
+++ b/mail/base/content/mailCore.js
@@ -624,11 +624,15 @@ function openAddonsMgr(aView) {
   });
 }
 
 function openActivityMgr() {
-  Cc["@mozilla.org/activity-manager-ui;1"]
-    .getService(Ci.nsIActivityManagerUI)
-    .show(window);
+  if (Services.prefs.getBoolPref("mail.activity.managerInTab", true)) {
+    openContentTab("chrome://messenger/content/activity.xhtml");
+  } else {
+    Cc["@mozilla.org/activity-manager-ui;1"]
+      .getService(Ci.nsIActivityManagerUI)
+      .show(window);
+  }
 }
 
 /**
  * Open the folder properties of current folder with the quota tab selected.
